asr1k_neutron_l3/models/neutron/l3/bgp.py

```diff
@@ -21,17 +21,22 @@
 from asr1k_neutron_l3.models.neutron.l3 import base
 
 
-class AddressFamily(base.Base):
-    def __init__(self, vrf, asn=None, routable_interface=False, rt_export=[],
-                 connected_cidrs=[], routable_networks=[], extra_routes=[]):
-        super(AddressFamily, self).__init__()
+class AddressFamilyBase(base.Base):
+    YANG_BGP_CLASS = None
+    YANG_BGP_NETWORK_CLASS = None
+
+    def __init__(self, vrf, asn=None, has_routable_interface=False, enable_af=True, rt_export=[],
+                 redistribute_map=None, connected_cidrs=[], routable_networks=[], extra_routes=[]):
+        super().__init__()
         self.vrf = utils.uuid_to_vrf_id(vrf)
-        self.routable_interface = routable_interface
+        self.enable_af = enable_af
+        self.has_routable_interface = has_routable_interface
         self.asn = asn
         self.enable_bgp = False
         self.rt_export = rt_export
+        self.redistribute_map = redistribute_map
         self.routable_networks = routable_networks
-        self.networks_v4 = set()
+        self.networks = set()
```

Comment:
any reason why this is not a list from the start?

Reply:
Deduplication / distrust of the input parameters we get. Just to make sure there are no duplicate networks in there, which would result in yang trying to reconfigure the same object again and again, as "my lists looks different than what's on the device".